spec/deserialization_spec.rb

```diff
@@ -2,85 +2,284 @@
 
 RSpec.describe JSONAPI::Deserialization do
   let(:jsonapi_deserialize) { UsersController.new.method(:jsonapi_deserialize) }
-  let(:document) do
-    {
-      data: {
-        id: 1,
-        type: 'note',
-        attributes: {
-          title: 'Title 1',
-          date: '2015-12-20'
-        },
-        relationships: {
-          author: {
-            data: {
-              type: 'user',
-              id: 2
-            }
-          },
-          second_author: {
-            data: nil
+
```

Comment:
This deserialization spec is pretty long now and I'm sure it could be DRYed out a fair amount (shared examples look promising though most of the length is specifying the data), but wanted to follow existing convention as it was.